pymeasure/instruments/redpitaya/redpitaya_scpi.py

```diff
@@ -0,0 +1,329 @@
+import datetime
+import time
+
+import numpy as np
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import modular_range, truncated_discrete_set, truncated_range, strict_discrete_set
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class DigitalChannelP(Channel):
+    """ A digital line of the P type"""
+
+    direction = Channel.control(
+        "DIG:PIN:DIR? DIO{ch}_P", "DIG:PIN:DIR %s,DIO{ch}_P",
+        """ Control a digital line to the given direction (str either 'IN' or 'OUT')""",
+        validator=truncated_discrete_set,
+        values=['IN', 'OUT'],
+    )
+
+    state = Channel.control(
+        "DIG:PIN? DIO{ch}_P", "DIG:PIN DIO{ch}_P,%d",
+        """ Control the state of the line (bool)""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, False: 0},
+    )
+
+
+class DigitalChannelN(Channel):
+    """ A digital line of the N type"""
+
+    direction = Channel.control(
```

Comment:
If you rename the property `direction_in`, and change it to accept booleans, it could be easier to handle than to give the proper string: `direction_in = True`, for example.

Reply:
done